homeassistant/components/pvpc_hourly_pricing/sensor.py

```diff
@@ -0,0 +1,234 @@
+"""
+Sensor to collect the reference daily prices of electricity ('PVPC') in Spain.
+
+For more details about this platform, please refer to the documentation at
+https://www.home-assistant.io/integrations/pvpc_hourly_pricing/
+"""
+from datetime import timedelta
+import logging
+from random import randint
+from typing import Optional
+
+from aiopvpc import PVPCData
+
+from homeassistant import config_entries
+from homeassistant.components.sensor import ENTITY_ID_FORMAT, PLATFORM_SCHEMA
+from homeassistant.const import CONF_NAME, CONF_TIMEOUT
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.entity import async_generate_entity_id
+from homeassistant.helpers.event import (
+    async_track_point_in_time,
+    async_track_time_change,
+)
+from homeassistant.helpers.restore_state import RestoreEntity
+import homeassistant.util.dt as dt_util
+
+from . import SENSOR_SCHEMA
+from .const import ATTR_TARIFF, DEFAULT_TIMEOUT, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(SENSOR_SCHEMA.schema)
+
+ATTR_PRICE = "price"
+ICON = "mdi:currency-eur"
+UNIT = "€/kWh"
+
+
+async def async_setup_platform(
+    hass: HomeAssistant, config, async_add_devices, discovery_info=None
+):
+    """
+    Set up the electricity price sensor as a sensor platform.
+
+    ```yaml
+    sensor:
+      - platform: pvpc_hourly_pricing
+        name: pvpc_manual_sensor
+        tariff: normal
+
+      - platform: pvpc_hourly_pricing
+        name: pvpc_manual_sensor_2
+        tariff: discrimination
+        timeout: 8
+    ```
+    """
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, data=config, context={"source": config_entries.SOURCE_IMPORT}
+        )
+    )
+    return True
+
+
+async def update_listener(hass: HomeAssistant, entry: config_entries.ConfigEntry):
+    """Update selected tariff for sensor."""
+    if entry.options[ATTR_TARIFF] != entry.data[ATTR_TARIFF]:
+        entry.data[ATTR_TARIFF] = entry.options[ATTR_TARIFF]
```

Comment:
Hi @bdraco,

I just rebased and now this writing on `entry.data` is not possible anymore. 
**How to transfer options data to entry for reload? **

Or should I remove the old one, use `asdict` + edition to generate new parameters, and call `setup_entry` instead?

Reply:
Ops, I think I have it now. I'm not using ok the data+options combo. I'll try to fix it